oop1-construct.js

```diff
@@ -1,16 +1,62 @@
+//Instantiate the class, and invoke the method so the results are showing in the console once this file is run.
 
 //your code here...
+class WhyClass {
+    constructor(purpose) {
+
+        this.purpose = purpose;
+    }
+    explain(purpose) {
+        console.log(purpose);
+    }
+    pieces() {
+        console.log('The important pieces of a class are the contructor, getters, setters, methods and attributes.')
+    }
+
 
 
+}
+
+let example = new WhyClass("a class is a blueprint for creating objects (a particular data structure), providing initial values for state (member variables or attributes), and implementations of behavior (member functions or methods). The user-defined objects are created using the class keyword.");
+
+example.explain(example.purpose);
+example.pieces();
 
-/*2. You are exploring the rainforests of the Amazon.  You have observed many different types of wildlife on your excursion.  To jog your memory, some different types of wildlife that you observed are found in the following URL : https://www.ietravel.com/blog/amazon-rainforest-animals-beginners-guide-21-species.  Feel free to use other resources online if you wish.
 
-Create a class that can categorize at least 6 of the creatures that you recall observing the Amazon.  Each creature should have 3 unique characteristics, and 2 unique actions that it can do. */
+/*2. You are exploring the rainforests of the Amazon.  You have observed many different types of wildlife on your excursion.  
+//To jog your memory, some different types of wildlife that you observed are found in the following URL :
+// https://www.ietravel.com/blog/amazon-rainforest-animals-beginners-guide-21-species.  
+//Feel free to use other resources online if you wish.
+
+Create a class that can categorize at least 6 of the creatures that you recall observing the Amazon.  
+//Each creature should have 3 unique characteristics, and 2 unique actions that it can do. */
 
 
 //your code here...
+class Mammal {
+    constructor(name, color, food, transit, noise) {
+        name;
+        color;
+        food;
+        transit;
+        noise;
+    }
+    move(transit) {
+        console.log(`${transit} is how I get from place to place`)
+    }
+    saySound(noise) {
+        console.log(`I go ${noise} !!!! `)
+    }
 
+}
+let dolphin = new Mammal("Amazon River dolphin", "pink", "fish", "swimming", "muaauuauau");
+let capybara = new Mammal("Capybara", "brown", "grass", "burrowing", "nonknonk");
+let armadillo = new Mammal("Giant Armadillo", "gray", "insects", "burrowing", "neek neek");
+let otter = new Mammal("Giant River Otter", "brown", "fish", "swimming", "cheecheechee");
+let jaguar = new Mammal("Jaguar", "Gold and Black", "the other mammals in this list", "stalking", "silence");
+let bat = new Mammal("Lophostoma Yasuni Bat", "black", "insects", "flying", "flipflap");
 
 
```

Comment:
do some stuff with your objects down here so I can see it working

`otter.saySound()`